src/utils/edlRequest.js

```diff
@@ -0,0 +1,106 @@
+import axios from 'axios'
+import snakeCaseKeys from 'snakecase-keys'
+
+import { downcaseKeys } from './downcaseKeys'
+import { pickIgnoringCase } from './pickIgnoringCase'
+import { prepKeysForCmr } from './prepKeysForCmr'
+import { buildEdlGroupPath } from './buildEdlGroupPath'
+
+/**
+ * Make a request to EDL and return the promise
+ * @param {Object} params
+ * @param {Object} params.headers Headers to send to CMR
+ * @param {String} params.method Method to make the request with
+ * @param {Object} params.params Parameters to send to CMR
+ * @param {Object} params.url Request URL
+ */
+export const edlRequest = ({
+  headers,
+  method,
+  params,
+  pathType,
+  requestInfo
+}) => {
+  let { params: requestParams } = params
+
+  // For mutations, params are not nested
+  if (!requestParams) requestParams = params
+
+  // Default headers
+  const defaultHeaders = {}
+
+  // Merge default headers into the provided headers and then pick out only permitted values
+  const permittedHeaders = pickIgnoringCase({
+    ...defaultHeaders,
+    ...headers
+  }, [
+    'Accept',
+    'Authorization',
+    'Client-Id',
+    'CMR-Request-Id',
```

Comment:
EDL isn't going to accept `CMR` prefixed headers. I'd send `X-Request-Id` which is the globally accepted version of `CMR-Request-Id` in the event that EDL handles it but their documentation doesn't suggest that they do anything with it.